src/extension.ts

```diff
@@ -211,7 +211,68 @@ function configureExpandMacro(context: ExtensionContext) {
     );
     panel.webview.html = getExpandMacroWebviewContent(res);
   });
-  
+
+  context.subscriptions.push(disposable);
+}
+
+interface ManipulatePipesResponseEdit {
+  newText: string,
+  range: {
+    start: {line: number, character: number},
+    end: {line: number, character: number}
+  }
+}
+
+interface ManipulatePipesResponse {
+  label: string
+  edit: {
+    changes: Record<string, ManipulatePipesResponseEdit[]>
+  }
+}
+
+function configureManipulatePipes(context: ExtensionContext, operation: "toPipe" | "fromPipe") {
+  const commandName = `extension.${operation}`;
+
+  const disposable = vscode.commands.registerCommand(commandName, async () => {
+    const extension = vscode.extensions.getExtension("jakebecker.elixir-ls");
+    const editor = vscode.window.activeTextEditor;
+    if (!extension || !editor) {
+      return;
+    }
+
+    const uri = editor.document.uri;
```

Comment:
This client retrieval is a common part of custom commands and could be extracted into a function